desktop/apps/auth/client/auth.coffee

```diff
@@ -0,0 +1,41 @@
+_ = require 'underscore'
+qs = require 'querystring'
+Backbone = require 'backbone'
+{ parse } = require 'url'
+{ API_URL } = require('sharify').data
+Form = require '../../../components/mixins/form.coffee'
+LoggedOutUser = require '../../../models/logged_out_user.coffee'
+
+module.exports.SignUpView = class SignUpView extends Backbone.View
+  _.extend @prototype, Form
+
+  events:
+    'submit': 'submit'
+
+  initialize: ->
+    @user = new LoggedOutUser
+
+  onRegisterSuccess: =>
+    window.location.href = @afterAuthHref
+
+  submit: (e) ->
+    return unless @validateForm()
+    return if @formIsSubmitting()
+
+    e.preventDefault()
+
+    action = $(e.target).data('action')
+    @afterAuthHref = $(e.target).data('after-auth')
+
+    @user.set (data = @serializeForm())
+    @user[action]
```

Comment:
We can almost use the same little form view here for both signup and login. Use the template to keep which 'action' it is.